### PR TITLE
User Management | Implement default avatars #4798

### PR DESCRIPTION
**Pull request by sergiogcx**
Closes https://github.com/cityofaustin/atd-data-tech/issues/4798

PR URL:
https://deploy-preview-202--atd-moped-main.netlify.app/

### CODE REVIEW

File: moped-editor/src/auth/user.js [hunk 1]
```diff
@@ -95,11 +95,25 @@ export const UserProvider = ({ children }) => {
   return <UserContext.Provider value={values}>{children}</UserContext.Provider>;
 };
 
+export const getRandomColor = () => {
+  const key = "atd_moped_user_color";
+  if (localStorage.getItem(key) === null) {
+    const randomInt = Math.floor(Math.random() * Math.floor(3));
+    const colors = ["#43a047", "#3949ab", "#fb8c00", "#e53935"];
```
Comment (mateoclarke) [inline moped-editor/src/auth/user.js:102]: should we use colors based on our Material Theme? Or at least store these hex codes as color variables in a constants file for re-use?

File: moped-editor/src/layouts/DashboardLayout/TopBar.js [hunk 2]
```diff
@@ -26,7 +25,21 @@ const useStyles = makeStyles(() => ({
...
+  const { user, logout } = useUser();
+
+  const emailToInitials = email => {
+    try {
+      const subdomain = "austintexas.gov";
+      if (!email.endsWith(subdomain)) return null;
+      const [first, last] = email
+        .replace("azure_ad", "")
+        .replace(subdomain, "")
+        .split(".");
```
Comment (mateoclarke) [inline moped-editor/src/layouts/DashboardLayout/TopBar.js:37]: Should we also capitalize the string to match the example in the issue?

**Review by mateoclarke — CHANGES_REQUESTED** [review]
All this looks nice. A few comments about formatting and where color hexes should live.

Also I think we missed another place where the Avatar icon lives: `/moped/account`.

Screen Shot 2021-02-08 at 3 59 10 PM

File: moped-editor/src/auth/user.js (hunk 1 above)
Comment (sergiogcx) [reply moped-editor/src/auth/user.js:102]: I took those from our material UI theme, I can re-work these as variables in a config file

File: moped-editor/src/layouts/DashboardLayout/TopBar.js (hunk 2 above)
Comment (sergiogcx) [reply moped-editor/src/layouts/DashboardLayout/TopBar.js:37]: Ah, good call! Mine was running capitalized, but I don't know why

File: moped-editor/src/auth/user.js (hunk 1 above)
Comment (mateoclarke) [reply moped-editor/src/auth/user.js:102]: Yeah, I think we should either use hexcode from or add to the `theme.palette` object or even make a new palette.js or color.js file in the `/theme` directory.

File: moped-editor/src/layouts/DashboardLayout/TopBar.js (hunk 2 above)
Comment (amenity) [reply moped-editor/src/layouts/DashboardLayout/TopBar.js:37]: Yes, capitalize plzzz! 🧑‍🏫

**Comment by amenity** [comment]
@sergiogcx - This feature was such a sweet surprise! Thanks for taking a sec to globalize what you implemented in the Activity Feed. 😊  🙏

**Review by amenity — APPROVED** [review]
Looks great! Mateo covered my notes — capitalization and the _Account_ page instance.

File: moped-editor/src/layouts/DashboardLayout/TopBar.js [hunk 3]
```diff
@@ -26,7 +25,21 @@ const useStyles = makeStyles(() => ({
...
+      return first.charAt(0) + last.charAt(0);
+    } catch {
```
Comment (johnclary) [inline moped-editor/src/layouts/DashboardLayout/TopBar.js:39]: cool to see try/catch here. i don't consider it enough w/ javascript

File: moped-editor/src/layouts/DashboardLayout/TopBar.js (hunk 2 above)
Comment (johnclary) [reply moped-editor/src/layouts/DashboardLayout/TopBar.js:37]: @sergiogcx I take it that a user with a non-austintexas.gov email address will have a colored circle with no initials? that's cool w/ me.

Comment (sergiogcx) [reply moped-editor/src/layouts/DashboardLayout/TopBar.js:37]: @johnclary Ah, sorry for the late reply. A user without initials would see a generic humanoid figure that is default fallback to Material UI:
https://material-ui.com/components/avatars/#fallbacks

**Comment by amenity** [comment]
LSTM 👌🏻 ✨

**Comment by amenity** [comment]
Not sure if it's because I'm testing with my non-COACD credentials, but the avatar on the account page is not showing my initials. 
Screen Shot 2021-02-10 at 12 08 43 PM

**Comment by sergiogcx** [comment]
It's not supposed to, it always defaults to null. I can change it real quick to show initials too, though the font size will be large.

**Comment by mateoclarke** [comment]
@sergiogcx I agree with Amenity. I think consistency with the avatar would be better.

**Comment by sergiogcx** [comment]
@amenity @mateoclarke there is a new PR:
https://github.com/cityofaustin/atd-moped/pull/215

**Pull request merged**